_posts/2020-12-16-solidity-0.7.6-release-announcement.md

```diff
@@ -0,0 +1,142 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.7.6 Release Announcement'
+date: '2020-12-16'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.7.6](https://github.com/ethereum/solidity/releases/tag/v0.7.6) adds better support for calldata types. Furthermore, the fallback function can now have a parameter and explicitly return data.
+
+## Notable New Features
+
+### Detection of Overflowing Unicode Direction Markers
+
+The winning entry in the
+[Solidity Underhanded Contest](https://blog.soliditylang.org/2020/12/03/solidity-underhanded-contest-winners/)
+used a clever combination of Unicode direction markers to deceive the reader about
+the semantics of a time-locked upgrade mechanism. Starting from Solidity 0.7.6, the parser will reject
+comments and Unicode strings where the text direction is not properly reset before the end of the
+comment or string.
+
+### Better Support for Calldata Types
+
+Since Solidity 0.6.9 you can use ``calldata`` as a data location specifier for variables other than
+parameters of external functions. This release finally adds important missing internal routines that
+make this feature even more useful:
+
+ - copying dynamically encoded structs and nested arrays from calldata to memory
+ - copying array slices from calldata to memory and storage
```

Comment:
Small indentation problem here. This gets handled correctly by github but other markdown parsers might treat it as a verbatim block because of this.
```suggestion
- copying dynamically encoded structs and nested arrays from calldata to memory
- copying array slices from calldata to memory and storage
```

Reply:
It looks fine in the preview, but I'll still commit it! https://deploy-preview-117--solidity-blog.netlify.app/2020/12/16/solidity-0.7.6-release-announcement/

Reply:
If it works on the blog then it's fine. Still, better to remove it just in case.